src/libproc/proc_pid.rs

```diff
@@ -229,15 +227,16 @@ pub fn get_errno_with_message(ret: i32) -> String {
 /// }
 /// ```
 pub fn listpids(proc_types: ProcType) -> Result<Vec<u32>, String> {
-    let mut pids: Vec<u32> = Vec::with_capacity(PID_MAX);
-    let buffer_ptr = pids.as_mut_ptr() as *mut c_void;
-    let buffer_size = (pids.capacity() * 4) as u32;
-    let ret: i32;
-
-    unsafe {
-        ret = proc_listpids(proc_types as u32, 0, buffer_ptr, buffer_size);
+    let buffer_size = unsafe { proc_listpids(proc_types as u32, 0, ptr::null_mut(), 0) };
+    if buffer_size <= 0 {
+        return Err(get_errno_with_message(buffer_size))
     }
 
+    let capacity = buffer_size as usize / mem::size_of::<u32>();
+    let mut pids: Vec<u32> = Vec::with_capacity(capacity);
+    let buffer_ptr = pids.as_mut_ptr() as *mut c_void;
+
+    let ret = unsafe { proc_listpids(proc_types as u32, 0, buffer_ptr, buffer_size as u32) };
     let items_count = ret as usize / mem::size_of::<u32>() - 1;
```

Comment:
We could move this into the else below - but pretty trivial point.

Reply:
@andrewdavidmackenzie
No problem, by the way, I'm not sure it should be `-1` but I can't check it, because I don't have a mac, and working with a VM takes too much time which I currently don't have. It was like this before, so can be investigated as a separate issue.